project/app/main.py

```diff
@@ -1,20 +1,250 @@
+        doc = nlp(text.lower())
+        ents = [e.text for e in doc.ents if e.label_ == 'GPE']
+        tokens_list.append(ents)
+    df['tokens'] = tokens_list
+
+    # figure out which city and state the article takes place in
+    city_list = []
+    state_list = []
+    geo_list = []
+    for tokens in df['tokens']:
+        # set up Counter
+        c = Counter(tokens)
+
+        # set up geolocation dict for geo list
+        geo_entry = {'lat': None, 'long': None}
+
+        # count which states come back the most, if any
+        state_counts = {}
+        for state in states_map:
+            if c[state] > 0:
+                state_counts[state] = c[state]
+
+        # get state(s) that came back the most as dict with lists
+        max_count = 0
+        max_state = None
+
+        for state in state_counts:
+            if state_counts[state] > max_count:
+                max_count = state_counts[state]
+                max_state = {state: {}}
+            elif state_counts[state] == max_count:
+                max_state[state] = {}
+
+        # if no state is found
+        if max_state is None:
+            city_list.append(None)
+            state_list.append(None)
+            geo_list.append(geo_entry)
+            continue
+
+        max_city = None
+        # get any cities in tokens based on states
+        for state in max_state:  # ideally this should only run once
+            city_counts = {}
+            for city in states_map[state]:
+                if c[city] > 0:
+                    city_counts[city] = c[city]
+            max_state[state] = city_counts
+
+            # get the city/state combo that came back the most
+            max_count = 0
+            for city in city_counts:
+                if city_counts[city] > max_count:
+                    max_count = city_counts[city]
+                    max_city = (city, state)
+
+        # if no city is found
+        if max_city is None:
+            city_list.append(None)
+            state_list.append(None)
+            geo_list.append(geo_entry)
+            continue
+
+        # the city and state should be known now
+
+        city_list.append(max_city[0].title())
+        state_list.append(max_city[1].title())
+        # now get the geolocation data
+        row = locs_df[(
+            (locs_df['city_ascii'] == max_city[0]) &
+            (locs_df['admin_name'] == max_city[1])
+        )]
+        row = row.reset_index()
+        if row.empty:
+            pass
+        else:
+            geo_entry['lat'] = row['lat'][0]
+            geo_entry['long'] = row['lng'][0]
+        geo_list.append(geo_entry)
+
+    # loop ends, add cities and states onto dataframe
+    df['city'] = city_list
+    df['state'] = state_list
+    df['geocoding'] = geo_list
+
+    # drop any columns with null entries for location
+    df = df.dropna()
+    df = df.reset_index()
+    df = df.drop(columns='index')
+
+    # cleanup to match 846 api
+    def listify(text):
```

Comment:
function definition is not detail oriented, what is the overall purpose of this function? Naming convention should be specific to its task.